src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1688,53 +1675,50 @@ internal static class Defaults
         /// <param name="initLearningRate">The initial learning rate used by SGD.</param>
         /// <param name="l2Weight">The L2 regularizer constant.</param>
         /// <param name="loss">The loss function to use.</param>
-        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
-        public StochasticGradientDescentClassificationTrainer(IHostEnvironment env,
+        internal StochasticGradientDescentClassificationTrainer(IHostEnvironment env,
```

Comment:
>internal StochasticGradientDescentClassificationTraine [](start = 8, length = 54)

I'd remove all ctors but the one with the (IHostEnvironment env, Arguments) signature, unless is needed for inheritance.

Reply:
exactly. thats the point of issue #2100.  but that would be a separate PR.   

these PRs focus on public API  which is P0  (as per consensus in email thread)

---
In reply to: [248772619](https://github.com/dotnet/machinelearning/pull/2093#discussion_r248772619) [](ancestors = 248772619)